apis/secretmanager/v1beta1/secretversion_types.go

```diff
@@ -28,24 +28,35 @@ type SecretManagerSecretVersionSpec struct {
 	// The resource name of the [Secret][google.cloud.secretmanager.v1.Secret] to create a [SecretVersion][google.cloud.secretmanager.v1.SecretVersion] for.
 	SecretRef *SecretRef `json:"secretRef,omitempty"`
 
-	// Immutable. The SecretVersion name. If not given, the metadata.name will be used.
+	// The SecretVersion number. If given, Config Connector acquires the resource from the Secret Manager service.
+	// If not given, Config Connector adds a new secret version to the GCP service, and you can find out the version number
+	// from `status.observedState.version`
 	ResourceID *string `json:"resourceID,omitempty"`
 
-	// TODO: Below fields are legacy KCC API. We should mark as deprecated once switch to use SciFi controller.
+	// Should enable or disable the current SecretVersion.
+	// - Enabled version can be accessed and described.
+	// - Disabled version cannot be accessed, but the secret's contents still exist
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// The actual secret data. Config Connector supports secret data stored in Kubernetes secret or plain data (base64)
+	SecretData *SecretData_OneOf `json:"secretData,omitempty"`
 
+	// DEPRECATED. You do not need to set this field in direct reconciler mode. Use delete-policy annotation instead. https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources#keeping_resources_after_deletion
 	// The deletion policy for the secret version. Setting 'ABANDON' allows the resource
 	// to be abandoned rather than deleted. Setting 'DISABLE' allows the resource to be
 	// disabled rather than deleted. Default is 'DELETE'. Possible values are:
 	// * DELETE
 	// * DISABLE
 	// * ABANDON.
 	DeletionPolicy *string `json:"deletionPolicy,omitempty"`
-	// The current state of the SecretVersion.
-	Enabled *bool `json:"enabled,omitempty"`
-	// Immutable. If set to 'true', the secret data is expected to be base64-encoded string and would be sent as is.
+
+	// DEPRECATED. You do not need to set this field in direct reconciler mode.
 	IsSecretDataBase64 *bool `json:"isSecretDataBase64,omitempty"`
-	// Immutable. The secret data. Must be no larger than 64KiB.
```

Comment:
The GCP SecretData is immutable, but KCC should. So that KCC can be used to acquire an existing resource that users do not want to store the sensitive data in k8s or via kcc.